art/defences/preprocessor/thermometer_encoding.py

```diff
@@ -155,6 +159,10 @@ def estimate_gradient(self, x: np.ndarray, grad: np.ndarray) -> np.ndarray:
         grad = grad * thermometer_grad
         grad = np.reshape(grad, grad.shape[:-1] + (grad.shape[-1] // self.num_space, self.num_space))
         grad = np.sum(grad, -1)
+
+        if self.channels_first:
+            grad = np.swapaxes(grad, 1, -1)
```

Comment:
How about using `permute` which preserves the ordering of H and W instead of `swapaxes`?
```suggestion
            grad = grad.permute(0, 3, 1, 2)
```